src/couch_changes.erl

```diff
@@ -336,6 +343,16 @@ get_doc_ids(_) ->
     throw({bad_request, no_doc_ids_provided}).
 
 
+get_selector({json_req, {Props}}) ->
+    check_selector(couch_util:get_value(<<"selector">>, Props));
+get_selector(#httpd{method='POST'}=Req) ->
+    couch_httpd:validate_ctype(Req, "application/json"),
+    {Props} = couch_httpd:json_body_obj(Req),
+    check_selector(couch_util:get_value(<<"selector">>, Props));
+get_selector(_) ->
+    throw({bad_request, invalid_selector}).
```

Comment:
The spec (obviously) is not so clear. We could send 405 Method Not Allowed but that's a resource-level thing and seems inappropriate. I'd send a 400 with a clear message if ?selector param exists, as both kxepal and davisp suggest.